core/arch/arm/mm/core_mmu.c

```diff
@@ -379,6 +379,10 @@ void core_mmu_set_discovered_nsec_ddr(struct core_mmu_phys_mem *start,
 		carve_out_phys_mem(&m, &num_elems, pmem->addr, pmem->size);
 #endif
 
+#ifdef CFG_DT
+	carve_out_phys_mem(&m, &num_elems, CFG_TZDRAM_START, CFG_TZDRAM_SIZE);
+#endif
```

Comment:
I think this should be made even without `CFG_DT`, for DDR ranges registered from `register_ddr()`.


Maybe also carveout `CFG_TZSRAM_START`/`CFG_TZSRAM_SIZE`. It can be useful when one wants to test the pager feature and uses a bit of DDR to emulate an small secure RAM.

By the way, prefer `if (IS_ENABLED(xxx))`. Or maybe this should be done in another P-R... later when one has time for that?

Reply:
+1

Reply:
Looks like with `CFG_TZSRAM` the build is broken. I could `#ifdef` that, `IS_ENABLED` is not really usuable here since the define may be missing entirely.

Reply:
Ok, forget about `CFG_TZSRAM_START`.
If one wants to enable pager without a specific TZSRAM then **generic_ram_layout.h** emulates a TZSRAM within TZDRAM:
```c
 * ----------------------------------------------------------------------------
 * TEE RAM layout with CFG_WITH_PAGER=y and undefined CFG_TZSRAM_START/_SIZE
 *
 *  +----------------------------------+ <-- CFG_TZDRAM_START
 *  | TEE core secure RAM (TEE_RAM)    |   | | CFG_CORE_TZSRAM_EMUL_SIZE
 *  +----------------------------------+ --|-'
 *  |   reserved (for kasan)           |   | TEE_RAM_VA_SIZE
 *  +----------------------------------+ --'
 *  | TA RAM / Pagestore (TA_RAM)      |
 *  +----------------------------------+ <---- align with CORE_MMU_PGDIR_SIZE
 *  +----------------------------------+ <--
 *  | SDP test memory (optional)       |   | CFG_TEE_SDP_MEM_SIZE
 *  +----------------------------------+ <-+ CFG_TZDRAM_START + CFG_TZDRAM_SIZE
 *
 *  +----------------------------------+ <-- CFG_SHMEM_START
 *  | Non-secure static SHM            |   |
 *  +----------------------------------+   v CFG_SHMEM_SIZE
```